pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -526,38 +544,50 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 		// persistently.
 		//
 		// Claims are treated as "allocated" if they are in the assume cache
-		// or currently their allocation is in-flight.
-		allocator, err := structured.NewAllocator(ctx, allocateClaims, &claimListerForAssumeCache{assumeCache: pl.claimAssumeCache, inFlightAllocations: &pl.inFlightAllocations}, pl.classLister, pl.sliceLister)
+		// or currently their allocation is in-flight. This does not change
+		// during filtering, so we can determine that once.
+		allocatedDevices := pl.listAllAllocatedDevices(logger)
```

Comment:
Moving `listAllAllocatedDevices` into the observer doesn't work because the observer doesn't know about `pl.inFlightAllocations`. But the name clash is indeed a bit odd. I've renamed the variable here to `allAllocatedDevices`.

Reply:
I was thinking about adding inFlightAllocation to the Lister, so in fact go into direction of dra_manager.go, but I guess it will be done this way in https://github.com/kubernetes/kubernetes/pull/127904 anyway.

The rename is good enough.